bno055/bno055.py

```
@@ -95,37 +90,41 @@ def main(args=None):
         lock = threading.Lock()
 
         def read_data():
-            """Callback for periodic data_query_timer executions in order to retrieve sensor IMU data"""
+            """Periodic data_query_timer executions to retrieve sensor IMU data."""
             if lock.locked():
-                # critical area still locked - that means that the previous data query is still being processed
-                node.get_logger().warn("Message communication in progress - skipping query cycle")
+                # critical area still locked
+                # that means that the previous data query is still being processed
+                node.get_logger().warn('Message communication in progress - skipping query cycle')
                 return
 
-            # Acquire lock before entering critical area in order to prevent overlapping data queries
+            # Acquire lock before entering critical area to prevent overlapping data queries
             lock.acquire()
             try:
                 # perform synchronized block:
                 node.sensor.get_sensor_data()
-            except Exception as e:
-                node.get_logger().warn('Receiving sensor data failed with %s:"%s"' % (type(e).__name__, e))
+            except TransmissionException as e:
```

This is actually the only location in the stacktrace where we catch an exception in this stack . It should be a general one to get all of them not only the ones that we throw. We  never know what could happen - e.g. TypeErrors and so on

---

so I changed this because we were failing the flake8 tests that checks for bare try/excepts - [this is the flake8 rule](https://github.com/elijahandrews/flake8-blind-except) that is throwing the error without this change

---

But in these cases we need a general exception handling as this is the only exception handling. If an exception is not handled here, the error will be handled in an uncontrolled way.
Furthermore the error handling at this location is the same for all errors: they will be logged but no more, so that the next cycle can be processed. So here it makes no sense to add specific error handling per exception type.

---

ok i agree. after doing some digging i think we can set flake8 to ignore specific lines of code for the `blind-except` errors. ill see if I can implement it and push an update here.

---

great, thx for investigating, Evan!